### DISCUSSION TITLE
Any progress info on SQL statements in traces?

### DISCUSSION THREAD

**Discussion by Temikus** [prisma/prisma#12081] [Q&A] [answered]
Hello folks 👋 

We're very much enjoying using prisma and really want to see SQL statements in traces.

There's a WIP note on the roadmap here but not a lot of details about when to expect it or if it has been started: https://www.notion.so/Support-Tracing-in-Prisma-Client-c0afd7599022435da2b8fc05c30cd1c0

Is there some idea on when this is expected to ship. Any info is good, even if it's something like "n(Quarters)" or "this year".

Thanks! 😄 

EDIT: Forgot to CC the assignee - @matthewmueller

**Comment by matthewmueller** [comment, marked as answer]
Hey @Temikus, thanks for reaching out! We're currently working on this bit-by-bit since MongoDB support has been our main focus.  We're aiming to have the basics up and running by early Q2 this year.

I'll plan on reaching back out to you for feedback once we have something to share 🙂

**Reply by Temikus** [reply]
Excellent, thank you for the answer ❤️

**Reply by superzadeh** [reply]
Also interested about this; even better if it enables us to integrate it with Datadog so we can see SQL statement in APM traces.

**Reply by jcmackie** [reply]
Any updates on this?

**Reply by matthewmueller** [reply]
Hey @Temikus @superzadeh @jcmackie! Super late reply, but I wanted to share that we're finally shipping Prisma Client tracing! 

Here's an early sneak peak of what's going out in our 4.2.0 release next Tuesday: https://deploy-preview-3408--prisma2-docs.netlify.app/docs/concepts/components/prisma-client/opentelemetry-tracing. 

You can already use Tracing in today in 4.1.0 though!

I'd love to get your feedback on our design and hit me up if you give tracing a shot in your app!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
